src/transformStateWithClones.js

```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const finalObject = [];
+  let modObject = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(modObject, action.extraData);
+      finalObject.push({ ...modObject });
+    }
+
+    if (action.type === 'removeProperties') {
+      action.keysToRemove.forEach((key) => delete modObject[key]);
+      finalObject.push({ ...modObject });
+    }
+
+    if (action.type === 'clear') {
+      modObject = {};
+      finalObject.push({ ...modObject });
+    }
+  }
+
+  return finalObject;
```

Comment:
There is no default case in your switch statement. It's a good practice to always include a default case to handle unexpected action types.